openfl-tutorials/interactive_api_tutorials_(experimental)/Pytorch_TinyImageNet_workspace/new_python_api_tinyiamgenet.ipynb

```diff
@@ -0,0 +1,704 @@
+    "        Data directory including `train` and `val` subdirectories.\n",
+    "    split: string\n",
+    "        Indicating which split to return as a data set.\n",
+    "        Valid option: [`train`, `val`]\n",
+    "    transform: torchvision.transforms\n",
+    "        A (series) of valid transformation(s).\n",
+    "    collabs: int\n",
+    "        How many dataset shards will be needed, minimum 1\n",
+    "    shard_num: int\n",
+    "        Current shard number, starting from 0\n",
+    "    \"\"\"\n",
+    "    def __init__(self, data='./data/tiny-imagenet-200/', is_validation=False, transform=None, target_transform=None):\n",
+    "        NUM_IMAGES_PER_CLASS = 500\n",
+    "        self.data = os.path.expanduser(data)\n",
+    "        self.transform = transform\n",
+    "        self.target_transform = target_transform\n",
+    "        split = 'train'\n",
+    "        if is_validation:\n",
+    "            split = 'val'\n",
+    "\n",
+    "        self.split_dir = os.path.join(self.data, split)\n",
+    "        self.image_paths = sorted(glob.iglob(os.path.join(self.split_dir, '**', '*.JPEG'), recursive=True))\n",
+    "        \n",
+    "        self.labels = {}  # fname - label number mapping\n",
+    "\n",
+    "        # build class label - number mapping\n",
+    "        with open(os.path.join(self.data, 'wnids.txt'), 'r') as fp:\n",
+    "            self.label_texts = sorted([text.strip() for text in fp.readlines()])\n",
+    "        self.label_text_to_number = {text: i for i, text in enumerate(self.label_texts)}\n",
+    "\n",
+    "        if split == 'train':\n",
+    "            for label_text, i in self.label_text_to_number.items():\n",
+    "                for cnt in range(NUM_IMAGES_PER_CLASS):\n",
+    "                    self.labels[f'{label_text}_{cnt}.JPEG'] = i\n",
+    "        elif split == 'val':\n",
+    "            with open(os.path.join(self.split_dir, 'val_annotations.txt'), 'r') as fp:\n",
+    "                for line in fp.readlines():\n",
+    "                    terms = line.split('\\t')\n",
+    "                    file_name, label_text = terms[0], terms[1]\n",
+    "                    self.labels[file_name] = self.label_text_to_number[label_text]\n",
+    "                    \n",
+    "    \n",
+    "    def __len__(self):\n",
+    "        return len(self.image_paths)\n",
+    "\n",
+    "    def __getitem__(self, index):\n",
+    "        file_path = self.image_paths[index]\n",
+    "        label = self.labels[os.path.basename(file_path)]\n",
+    "        label = self.target_transform(label) if self.target_transform else label\n",
+    "        return self.read_image(file_path), label\n",
+    "\n",
+    "    def read_image(self, path):\n",
+    "        img = Image.open(path)\n",
+    "        return self.transform(img) if self.transform else img\n",
+    "\n",
+    "def one_hot(labels, classes):\n",
+    "    return np.argmax(np.eye(classes)[labels])"
```

Comment:
`one_hot` function expected to return a one-hot tensor representation of the `labels`. But this definition applies both forward and backward one-hot transformation, returning labels without modifying.